### ISSUE TITLE
Syncthing Conflicts Filenames with Upper Case Extensions with Lowercase Equivalents

### ISSUE THREAD

**Issue by xcjs** [syncthing/syncthing#8922]
### Include required information

Please be sure to include at least:

 - which version of Syncthing and what operating system you are using
   - v1.23.4 on Windows 10
   - v1.23.4 on Steam OS
 - browser and version, if applicable
   - N/A for this bug
 - what happened,
   - I have a few files with case sensitive file extensions that I use for a specific application. These file extensions must be upper case (filename.EXT). Once these files are created on a Linux based system (Steam OS in this case), Syncthing syncs these files with uppercase extensions (filename.EXT) to Windows 10, however Syncthing reports a conflict with versions of these files featuring a lowercase extensions as well, even including conflict files for the affected files. (.syncthing-filename.ext) and reports in the UI that the sync failed.
 - what you expected to happen instead, and
   - I know that Windows has a case insensitive filesystem, which is probably part of the problem, however Syncthing doesn't seem to recognize that these files already exist with the proper casing. If possible, I don't think Syncthing should create a conflict and report the sync as failed.
 - any steps to reproduce the problem.
   - I feel that there's enough information above to recreate the issue, but I can answer any additional questions you may have.

**Comment by calmh** [comment]
Steps to reproduce the problem, please? Syncthing tries to be case insensitive, like windows and macOS, by default.

**Comment by xcjs** [comment]
Steps to reproduce:

1. Create directory share on a Linux-based system.
2. Add a file or several files with a lowercase filename but uppercase file extensions (such as filename.EXT).
3. Add directory share on a Windows system.
4. Files will sync to Windows with the correct case.
5. Syncthing on Windows will report the directory as Out of Sync with conflict files in the Windows directory with lowercase file extensions.

You mentioned that Syncthing tries to be case insensitive, but is there a way to force it to be case sensitive?

**Comment by xcjs** [comment]
Ok, so for laughs I enabled the case sensitivity for NTFS on the directory in question. (As explained [here](https://learn.microsoft.com/en-us/windows/wsl/case-sensitivity).)

Syncthing has created both filename.ext _and_ filename.EXT. This has resolved the "Out of Sync" error, but I'm not sure why it would do this.

**Comment by calmh** [comment]
Yeah... None of that happens for me, nor should it, so I'm not sure what's going on in your case, sorry. Perhaps there's some other Windows software running that automatically tries to lower case the extensions?

<img width="1470" alt="Screenshot 2023-06-04 at 08 24 32" src="https://github.com/syncthing/syncthing/assets/125426/9378a4db-ef63-40b0-bf64-2f7e6b5564fa">

**Comment by xcjs** [comment]
I have no other software accessing or using the directory and am still having this issue, though the NTFS flag is somewhat of a work-around.

**Comment by calmh** [comment]
Yeah, I don't know what to tell you. I can't reproduce it, and we have tens of thousands of users on Windows syncing files, and no one else has reported seeing this -- and since it's fairly spectacular breakage I'm sure we'd have heard if it was systematic. It seems like something specific to your system.
